oss-internship-2020/libtiff/sandboxed.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <linux/futex.h>
+#include <syscall.h>
+
+#include <utility>
+
+#include "sandboxed_api/util/flag.h"
+#include "tiff_sapi.sapi.h"
+
+ABSL_DECLARE_FLAG(string, sandbox2_danger_danger_permit_all);
+ABSL_DECLARE_FLAG(string, sandbox2_danger_danger_permit_all_and_log);
+
+namespace {
+
+class TiffSapiSandbox : public TiffSandbox {
+ public:
+  explicit TiffSapiSandbox(std::string dir = "", std::string file = "")
```

Comment:
As mentioned before, check `std::optional<std::string>`. `explicit` is only needed if the constructor has only one argument.
Also you can consider if it is cleaner to have only one argument that is mandatory and then add a function to this class that will allow setting the second parameter.

E.g.
```
TiffSapiSandbox sandbox("/input/file.tiff");
sandbox.SetTheImportantDirectory("/some/path");
```

Reply:
I wasn't sure how to use explicit if the second parameter has a default value
Is SetTheImportantDirectory better than using std::optional by default?